### DISCUSSION TITLE
Has anyone managed to connect to storage in local using S3?

### DISCUSSION THREAD

**Discussion by samuelwhunter** [supabase/supabase#29243] [Questions]
I've been really struggling to connect to the storage routes in local with S3. I'm using Uppy to handle the front end and then pointing everything to the storage endpoint provided by Supabase local (see below).

S3 Storage URL: http://127.0.0.1:54321/storage/v1/s3

I keep getting a 404 No Found error though. I've set the S3Client up with the access and secret key that the CLI provides too, but nothing. Am I hitting the correct endpoint? It feels like I'm missing something but can't work it out.

I've been following these: 
https://supabase.com/blog/s3-compatible-storage
https://supabase.com/docs/guides/storage/s3/authentication

My thoughts are that it's just ready in local yet as S3 is in public alpha, but if that's the case I'm unsure why all the keys are shown when firing it up.

**Comment by platform-kit** [comment]
@samuelwhunter did you ever get anywhere with this? I've been unsuccesful at uploading to local supabase via S3 sdk too.

**Comment by activenode** [comment]
Here's the version for cyberduck, you should analogously be able to connect via any S3 client.

You use this and save it as `supabase.cyberduckprofile` :

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>Protocol</key>
        <string>s3</string>
        <key>Vendor</key>
        <string>Supabase Storage S3</string>
        <key>Scheme</key>
        <string>http</string>
        <key>Description</key>
        <string>Supabase Storage (S3)</string>
        <key>Default Hostname</key>
        <string>127.0.0.1</string>
        <key>Default Port</key>
        <string>54321</string>
        <key>Region</key>
        <string>local</string>
        <key>Properties</key>
        <array>
            <string>s3.storage.class.options=STANDARD</string>
            <string>s3.bucket.virtualhost.disable=true</string>
        </array>
        <key>Context</key>
        <string>/storage/v1/s3</string>
        <key>Help</key>
        <string>https://supabase.com/docs/guides/storage</string>
    </dict>
</plist>
```

(`region=local` is important!)

Then, double click it and provide the Access Key Id and Secret for S3 that (`npx supabase status`).

You're in 🥳 

Happy to help.
Your Supabase Guru, [activeno.de](https://activeno.de) :)

**Reply by MichaelDeBoey** [reply]
I keep getting "Unsupported or unrecognized SSL message" error for some reason, even though `scheme` is set to `http` 🤔

**Comment by aantti** [comment]
Just in case - I've recently added a couple of how-to's regarding [self-hosted Supabase](https://supabase.com/docs/guides/self-hosting) and using S3 client/S3 backend:

- https://supabase.com/docs/guides/self-hosting/self-hosted-s3
- https://supabase.com/docs/guides/self-hosting/copy-from-platform-s3
